Philips.CodeAnalysis.MaintainabilityAnalyzers/NamespacePrefixAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class NamespacePrefixAnalyzer : DiagnosticAnalyzer
+	{
+		#region Non-Public Data Members
+
+		private const string Title = @"Namespace must use the predefined prefixes";
+		private const string MessageFormat = @"Namespace must use the predefined prefixes configured in the editorconfig file ie. [OrganizationName].[ProductName]";
+		private const string Description = @"Namespace must use the predefined prefixes";
+		private const string Category = Categories.Naming;
+
+		#endregion
+
+		#region Non-Public Properties/Methods
+		private void Analyze(SyntaxNodeAnalysisContext context)
+		{
+
```

Comment:
Does this re-parse the .editorconfig each time a Namespace declaration is found?  Or does AdditionalFilesHelper cache?

Reply:
It re-parses the .editorconfig each time a Namespace declaration is found.